src/parse/text/simple_extensions/argument.rs

```diff
@@ -0,0 +1,512 @@
+                name: Some(name),
+                description: Some(description),
+                options,
+            },
+
+            ArgumentsItem::ValueArgument(ValueArgument {
+                name,
+                description,
+                value,
+                constant,
+            }) => simple_extensions::ArgumentsItem::Variant1 {
+                name: Some(name),
+                description: Some(description),
+                value,
+                constant: Some(constant),
+            },
+
+            ArgumentsItem::TypeArgument(TypeArgument {
+                name,
+                description,
+                type_,
+            }) => simple_extensions::ArgumentsItem::Variant2 {
+                name: Some(name),
+                description: Some(description),
+                type_,
+            },
+        }
+    }
+}
+
+/// Parse errors for [simple_extensions::ArgumentsItem].
+#[derive(Debug, Error, PartialEq)]
+pub enum ArgumentsItemError {
+    /// Empty enumeration option.
+    #[error("empty enumeration option")]
+    EmptyEnumOption,
+
+    /// Empty optional field.
+    #[error("the optional field `{0}` is empty and should be removed")]
+    EmptyOptionalField(String),
+}
+
+/// Arguments that support a fixed set of declared values as constant arguments.
+#[derive(Clone, Debug, PartialEq)]
+pub struct EnumArgument {
+    /// A human-readable name for this argument to help clarify use.
+    name: String,
+
+    /// Additional description for this argument.
+    description: String,
+
+    /// List of valid string options for this argument.
+    options: Vec<String>,
+}
+
+impl EnumArgument {
+    fn parse(
+        name: Option<String>,
+        description: Option<String>,
+        options: Vec<String>,
+    ) -> Result<EnumArgument, ArgumentsItemError> {
+        if options.iter().any(String::is_empty) {
+            return Err(ArgumentsItemError::EmptyEnumOption);
+        }
+
+        Ok(EnumArgument {
+            name: ArgumentsItem::parse_name(name)?,
+            description: ArgumentsItem::parse_description(description)?,
+            options,
+        })
+    }
+}
+
+impl From<EnumArgument> for ArgumentsItem {
+    fn from(value: EnumArgument) -> Self {
+        ArgumentsItem::EnumArgument(value)
+    }
+}
+
+/// Arguments that refer to a data value.
+#[derive(Clone, Debug)]
+pub struct ValueArgument {
+    /// A human-readable name for this argument to help clarify use.
+    name: String,
+
+    /// Additional description for this argument.
+    description: String,
+
+    /// A fully defined type or a type expression.
+    ///
+    /// todo: implement parsed [simple_extensions::Type].
+    value: simple_extensions::Type,
+
+    /// Whether this argument is required to be a constant for invocation.
+    /// For example, in some system a regular expression pattern would only be accepted as a literal
+    /// and not a column value reference.
+    constant: bool,
```

Comment:
As specified in doc, this defaults to false if absent. I think we can remove the option here?

Reply:
This would make the back-conversion lossy, however, this is fine because it doesn't change behaviour. And I guess it makes it easier for users to not have to be aware of this (default=false) detail.

Reply:
Make sense.